numba/core/dispatcher.py

```diff
@@ -983,6 +983,13 @@ def compile(self, sig):
             flags = self.flags
             args, return_type = sigutils.normalize_signature(sig)
 
+            # for objmode we should have ffi_forced_objects as args
+            # and no_rewrites flag can be set to avoid recompilation with and
+            # without rewrites in compile_ir
+            if isinstance(self, ObjModeLiftedWith):
+                args = [types.ffi_forced_object] * len(args)
+                flags.no_rewrites = True
+
```

Comment:
Would it work to put this logic into an override of the `compile` method in the `ObjModeLiftedWith` subclass, and have this method then call `super().compile()`, this keeping subclass details out of the base class? The flags can be set in the initializer too perhaps. Was thinking of something like:
```diff
diff --git a/numba/core/dispatcher.py b/numba/core/dispatcher.py
index 9b5b66b61..659964077 100644
--- a/numba/core/dispatcher.py
+++ b/numba/core/dispatcher.py
@@ -1063,6 +1063,8 @@ class ObjModeLiftedWith(LiftedWith):
             raise ValueError("expecting `flags.force_pyobject`")
         if self.output_types is None:
             raise TypeError('`output_types` must be provided')
+        # switch off rewrites, they have no effect
+        self.flags.no_rewrites = True
 
     @property
     def _numba_type_(self):
@@ -1107,6 +1109,12 @@ class ObjModeLiftedWith(LiftedWith):
                 )
                 raise errors.TypingError(msg.format(i))
 
+    @global_compiler_lock
+    def compile(self, sig):
+        args, _ = sigutils.normalize_signature(sig)
+        sig = (types.ffi_forced_object,) * len(args)
+        return super().compile(sig)
+
 ```
might this work?

Reply:
@stuartarchibald Yes, it certainly works and it's also much better from code organization perspective. Thanks!